### PR TITLE
chore: simplified `awk` usage for version extraction

### PR DESCRIPTION
**Pull request by fearphage**
# Proposed changes

Simplified version extract from Gradle files from 3 calls to 1. 

## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [ ] I checked that all workflows return a success.
- [x] I included all the needed documentation for this change.
- [x] I provided the necessary tests.
- [x] I squashed all the commits into a single commit.
- [x] I followed the [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I wrote the necessary upgrade instructions in the [upgrade guide](../docs/upgrade-guide.md).
- [ ] If this pull request is about and existing issue,
  I added the `Fix #ISSUE_NUMBER` or `Close #ISSUE_NUMBER` text to the description of the pull request.

### Super-linter maintainer tasks

- [x] Label as `breaking` if this change breaks compatibility with the previous released version.
- [x] Label as either: `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`.
- [x] Add the pull request to a milestone, eventually creating one, that matches with the version that release-please proposes.

### CODE REVIEW

File: docs/add-new-linter.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ new tool, it should include:
       You get the version of a dependency from `build.gradle`. Example:
 
         ```sh
-        GOOGLE_JAVA_FORMAT_VERSION="$(grep <"google-java-format/build.gradle" "google-java-format" | awk -F ':' '{print $3}' | tr -d "'")"
+        GOOGLE_JAVA_FORMAT_VERSION="$(awk -F "[:']" '/google-java-format/ {print $3}' "google-java-format/build.gradle")"
```
Comment (ferrarimarco) [inline docs/add-new-linter.md:58]: `print $3` -> `print $4`

Comment (ferrarimarco) [inline docs/add-new-linter.md:58]: Can you refactor this to:

```
GOOGLE_JAVA_FORMAT_VERSION="$(
  set -euo pipefail
  awk -F "[:']" '/google-java-format/ {print $4}' "google-java-format/build.gradle"
)"

**Review by ferrarimarco — CHANGES_REQUESTED** [review]
Thanks for this PR! A couple of small things to check.

File: docs/add-new-linter.md (hunk 1 above)
Comment (fearphage) [reply docs/add-new-linter.md]: Updated.

**Review by ferrarimarco — APPROVED** [review]

**Comment by ferrarimarco** [comment]
Thanks for this PR!

**Pull request merged**